### ISSUE TITLE
ephemeral_storage_local_ssd_config doesn't work

### ISSUE THREAD

**Issue by visortelle** [terraform-google-modules/terraform-google-kubernetes-engine#1999]
### TL;DR

Hi 👋

When I specify `ephemeral_storage_local_ssd_config = 1`, I don't see local disks attached to the created VMs in the GCP console. 
Also, I don't see anything related to `ephemeral_storage_local_ssd_config` after running `terraform plan`.

If I add a node pool using UI, it works properly and I see the NVMe disk in the local disks list of the created instance.

<img width="890" alt="Screenshot 2024-07-17 at 8 05 02 PM" src="https://github.com/user-attachments/assets/17a845ac-56b7-4d71-a809-7c02b32d4db7">

@alvicsam could you please help me with this issue?

Thanks.

### Terraform Configuration

```hcl
    {
      autoscaling                       = true
      local_ssd_ephemeral_storage_count = 1
      machine_type                      = "n2-standard-8"
      name                              = "xxx"
      node_locations                    = "europe-west3-b"
      spot                              = true
      total_max_count                   = 10
      total_min_count                   = 0
    }
```

### Terraform Version

```sh
Terraform v1.9.2
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v5.38.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.31.0
+ provider registry.terraform.io/hashicorp/random v3.6.2
```

**Comment by visortelle** [comment]
Sorry. Looks like I messed up with the provider versions. Closing.

**Comment by Ksamuel92** [comment]
@visortelle Hi! Stumbled across this because I’m running into the exact same issue. Would you mind elaborating on the provider versions comment you left?
